airbyte-integrations/connectors/destination-iceberg/src/test-integration/java/io/airbyte/integrations/destination/iceberg/hadoop/BaseIcebergHadoopCatalogS3IntegrationTest.java

```diff
@@ -103,9 +96,21 @@ protected List<JsonNode> retrieveRecords(final TestDestinationEnv testEnv,
                                            final String namespace,
                                            final JsonNode streamSchema)
       throws Exception {
-    return IcebergIntegrationTestUtil.retrieveRecords(getConfig(), namespace, streamName);
+    return IcebergIntegrationTestUtil.retrieveRecords(getConfigForTestRunner(), namespace, streamName);
   }
 
   abstract DataFileFormat fileFormat();
 
+  @Nullable
+  @Override
+  protected String getDefaultSchema(@NotNull JsonNode config) throws Exception {
+    // TODO: This was NPE'ing without this return value because of Kotlin's non-null in base,
+    // but whats the actual value to pass instead of empty ?
```

Comment:
make the function abstract?

Reply:
where are we inheriting this as nonnullable from? DAT.kt says it returns `String?` on master https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/DestinationAcceptanceTest.kt#L261

Reply:
Bad comment, actually was doing `!!` in actual tests for defaultSchema which was set as null in base class.